internal/generator/fri/template/fri.test.go.tmpl

```diff
@@ -0,0 +1,222 @@
+import (
+	"crypto/sha256"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/consensys/gnark-crypto/ecc/{{.Name}}/fr"
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/gen"
+	"github.com/leanovate/gopter/prop"
+)
+
+// logFiber returns u, v such that {g^u, g^v} = f⁻¹((g²)^{_p})
+func logFiber(_p, _n int) (_u, _v big.Int) {
+	if _p%2 == 0 {
+		_u.SetInt64(int64(_p / 2))
+		_v.SetInt64(int64(_p/2 + _n/2))
+	} else {
+		l := (_n - 1 - _p) / 2
+		_u.SetInt64(int64(_n - 1 - l))
+		_v.SetInt64(int64(_n - 1 - l - _n/2))
+	}
+	return
+}
+
+func randomPolynomial(size uint64, seed int32) []fr.Element {
+	p := make([]fr.Element, size)
+	p[0].SetUint64(uint64(seed))
+	for i := 1; i < len(p); i++ {
+		p[i].Square(&p[i-1])
+	}
+	return p
+}
+
+// convertOrderCanonical convert the index i, an entry in a
+// sorted polynomial, to the corresponding entry in canonical
+// representation. n is the size of the polynomial.
+func convertSortedCanonical(i, n int) int {
+	if i%2 == 0 {
+		return i / 2
+	} else {
+		l := (n - 1 - i) / 2
+		return n - 1 - l
+	}
+}
+
+func TestFRI(t *testing.T) {
+
+	parameters := gopter.DefaultTestParameters()
+	parameters.MinSuccessfulTests = 10
+
+	properties := gopter.NewProperties(parameters)
+
+	size := 4096
```

Comment:
remark; shouldn't his vary during the tests? 

Similarly, `randomPolynomial` is not super random; (ie always the same form with coeff squared)

Reply:
@ThomasPiellard